src/app.py

```diff
@@ -157,39 +156,40 @@ def job_statuses() -> JobsSchema:
 
 @app.get("/api/backend", response_model=BackendStatusSchema)
 def backend_status() -> BackendStatusSchema:
-    db_alive = True
-    status = ursa.status()
-    try:
-        tasks = status.get("result", {}).get("tasks", [])
-        ursadb_version = status.get("result", {}).get(
-            "ursadb_version", "unknown"
-        )
-    except Again:
-        db_alive = False
-        tasks = []
-        ursadb_version = []
-
-    return BackendStatusSchema(
-        db_alive=db_alive,
-        tasks=tasks,
-        components={
-            "mquery": mquery_version(),
-            "ursadb": str(ursadb_version),
-        },
-    )
+    agents = []
+    components = {
+        "mquery": mquery_version(),
+    }
+    for name, url in db.get_active_agents().items():
+        try:
+            ursa = UrsaDb(url)
+            status = ursa.status()
+            tasks = status["result"]["tasks"]
+            ursadb_version = status["result"]["ursadb_version"]
+            agents.append(
+                {"name": name, "alive": True, "tasks": tasks, "url": url}
+            )
+            components[f"ursadb ({name})"] = ursadb_version
+        except Again:
+            agents.append(
+                {"name": name, "alive": False, "tasks": [], "url": url}
+            )
+            components[f"ursadb ({name})"] = "unknown"
+
+    return BackendStatusSchema(agents=agents, components=components,)
 
 
 @app.get("/api/backend/datasets", response_model=BackendStatusDatasetsSchema)
 def backend_status_datasets() -> BackendStatusDatasetsSchema:
-    db_alive = True
-
-    try:
-        datasets = ursa.topology().get("result", {}).get("datasets", {})
-    except Again:
-        db_alive = False
-        datasets = {}
-
-    return BackendStatusDatasetsSchema(db_alive=db_alive, datasets=datasets)
+    datasets = {}
+    for url in db.get_active_agents().values():
+        try:
+            ursa = UrsaDb(url)
+            datasets.update(ursa.topology()["result"]["datasets"])
+        except Again:
+            pass
+
+    return BackendStatusDatasetsSchema(db_alive=True, datasets=datasets)
```

Comment:
`db_alive` is set to True even if there are no active agents - is it correct? Maybe that Schema field should be removed if it's no longer needed.

Reply:
Don't need to be stable if it's undocumented :smarts: